### PR TITLE
SDK-857: Handle empty attribute values

### PR DESCRIPTION
**Pull request by echarrod**

### CODE REVIEW

File: src/Yoti.Auth/AttributeConverter.cs [hunk 1]
```diff
@@ -36,6 +36,10 @@ public static BaseAttribute ConvertToBaseAttribute(ProtoBuf.Attribute.Attribute
         {
             NLog.Logger logger = NLog.LogManager.GetCurrentClassLogger();
 
+            if (attribute.ContentType != ContentType.String
+                    && attribute.Value.Length == 0)
+                throw new InvalidOperationException("Empty value is invalid for non-string content types");
```
Comment (davidgrayston) [inline src/Yoti.Auth/AttributeConverter.cs]: Could we add test coverage for this exception? Or perhaps this method should be private as it's only being used by `Yoti.Auth.Tests.ActivityTests::AddAttributeToProfile()`?

Comment (echarrod) [reply src/Yoti.Auth/AttributeConverter.cs]: In `OtherAttributesWithEmptyValueAreNotAdded` we're testing that each of the attribute types apart from string are causing the attribute to not be added when the empty value it attempted to be added. I could add further tests for this method though, to check that this exact exception is being thrown.

Comment (echarrod) [reply src/Yoti.Auth/AttributeConverter.cs]: Also the class is `internal`, so this method would only inherit the `internal` access modifier.

Comment (echarrod) [reply src/Yoti.Auth/AttributeConverter.cs]: @davidgrayston have added some further tests in a098202

**Pull request merged**